src/crimson/osd/object_context_loader.cc

```diff
@@ -116,6 +127,15 @@ using crimson::common::local_conf;
         }
         obc->set_head_state(std::move(md->os),
                             std::move(md->ssc));
+	for (auto &[src, winfo] : obc->obs.oi.watchers) {
+	  auto [it, emplaced] = obc->watchers.emplace(src, nullptr);
+	  assert(emplaced);
+	  logger().debug("adding watch for obj {}, client {}",
+	    obc->get_oid(), src.second);
+	  it->second = crimson::osd::Watch::create(
+	    obc, winfo, src.second, std::move(pg));
```

Comment:
In order to avoid passing a pg ref into obc_loader's methods we can try using the watcher's pg:
```
it->second = crimson::osd::Watch::create(
  obc, winfo, src.second, it->second->get_pg());
```
Another option is to handle this bit right after loading the obc in `PG::with_locked_obc()` for primaries only.

Also, let's move this into a separate function since more logic will be added soon (`check_blocklisted_obc_watchers`)

Reply:
will do